api/registrations/serializers.py

```diff
@@ -343,11 +343,14 @@ def update(self, registration, validated_data):
                     raise exceptions.PermissionDenied()
             else:
                 raise exceptions.ValidationError('Registrations can only be turned from private to public.')
+        if not user_is_admin and (validated_data.get('retraction') or validated_data.get('is_retracted')):
```

Comment:
We need to check if `is_retracted` or `retraction` is in the payload, regardless of the value, and if so, raise PermissionDenied().

If `is_retracted` was False, here for example, this PermissionDenied wouldn't be raised here.  I've gotten burned by this before.  Something like this should work:

 ```python
if 'retraction' in validated_data or 'is_retracted' in validated_data:
    ...
```